packages/@react-aria/utils/src/useId.ts

```diff
@@ -47,7 +47,15 @@ export function useId(defaultId?: string): string {
   }, [setValue, updateValue]);
 
   let res = useSSRSafeId(value);
+
   idsUpdaterMap.set(res, updateValue);
```

Comment:
Shouldn't this go in the useLayoutEffect as well? And the whole `updateValue` function too? This whole function feels a bit weird...

Reply:
Yeah, definitely a scary function now, to save us from breaking in React 17 since it'd try to update an id state to a component that was rendering
I tried moving it in and maybe I did something bad but I definitely broke some tests
I did realize that sometimes we set a key in the map as null, so that's not great, pushing a change for it now
I think we should revisit this function though, but we should pair on it